src/client/src/tscSQLParser.c

```diff
@@ -9142,7 +9153,9 @@ int32_t tscGetExprFilters(SSqlCmd* pCmd, SQueryInfo* pQueryInfo, SArray* pSelect
     SStrToken* pToken = &pParam->pNode->columnName;
 
     SColumnIndex idx = COLUMN_INDEX_INITIALIZER;
-    getColumnIndexByName(pToken, pQueryInfo, &idx, tscGetErrorMsgPayload(pCmd));
+    if (getColumnIndexByName(pToken, pQueryInfo, &idx, tscGetErrorMsgPayload(pCmd)) != TSDB_CODE_SUCCESS) {
```

Comment:
Check the retcode of getColumnIndexByName with an invalid column name for the having() function. Otherwise, an assertion violation happens.